docs/concepts.md

```diff
@@ -50,9 +53,325 @@ Dynamic provisioning is when an resource claim does not find a matching resource
+
+### Secure Connectivity
+networks, subnets, peering, NACLs,etc.
+
+# Glossary
+
+## Kubernetes
+Crossplane is built on the Kubernetes API machinery as a platform for declarative management,
+and common terminology used from the [Kubernetes Glossary][kubernetes-glossary] where possible,
+and we don't seek to reproduce that glossary here.
+
+[kubernetes-glossary]: https://kubernetes.io/docs/reference/glossary/?all=true
+However we'll summarize some commonly used concepts for convenience.
+
+### CRD
+A standard Kubernetes Custom Resource Definition (CRD), which defines a new type of resource that can be managed declaratively. 
+This serves as the unit of management in Crossplane.
+The CRD is composed of spec and status sections and supports API level versioning (e.g., v1alpha1)
+
+### Controller
+A standard Kubernetes Custom Controller, providing active control loops that own or more CRDs. 
+Can be implemented in different ways, such as 
+golang code (controller-runtime), templates, functions/hooks, templates, a new DSL, etc. 
+The implementation itself is versioned using semantic versioning (e.g., v1.0.4)
+
+### Namespace
+Allows logical grouping of resources in Kubernetes that can be secured with RBAC rules.
+
+## Crossplane
+### External CRD
+Usually represent an external resource, and cannot be broken down any further (leaves)
+
+aka Atomic CRD
+
+### Composite CRD
+A resource that captures parent/child relationships. They have a selector that can help query/find children resources.
+
+### Claim CRD
+An abstract resource that binds to managed resources, like a MySQLInstance..
+
+### App CRD
+An abstract resource that manages an application instance, like a WordpressInstance.
+
+### Stack
+Thhe unit of extending Crossplane with new functionality.  It is comprised of the CRDs, 
+a Custom Controller, and metadata about the Stack. 
+A Crossplane cluster can be queried for all installed Stacks as a way of learning 
+what functionality a particular Crossplane supports.
+
+### Stack Registry
+this is a registry for Stacks where they can be published, downloaded, explored, categorized, etc. 
+The registry understands a Stack’s custom controller and its CRDs and indexes by both -- you could lookup a custom controller by the CRD name and vice versa.
+
+### Stack Package Format
+The package format for Stacks that contains the Stack definition, metadata, icons, CRDs, and other Stack specific files.  
+
+### Stack Manager
+The component that is responsible for installing a Stack’s custom controllers and resources in Crossplane. It can download packages, resolve dependencies, install resources and execute controllers.  This component is also responsible for managing the complete life-cycle of Stacks, including upgrading them as new versions become available.
+
+### External Resource
+An actual resource that exists outside Kubernetes, typically in the cloud. 
+AWS RDS and GCP Cloud Memorystore instances are external resources.
```

Comment:
I've been trying to say something like "in an external system, often a cloud provider" in an attempt to suggest that external resources could be on prem, while also providing some slightly less abstract/hypothetical context.